src/nodes.coffee

```diff
@@ -3639,6 +3639,8 @@ exports.Assign = class Assign extends Base
     # destructured variables.
     @variable.base.propagateLhs yes
 
+  isStatementAst: NO
```

Comment:
Need to override the default `isStatement()` logic for `Assign` so that the AST always treats assignments as non-statements and adds an `ExpressionStatement` wrapper